### DISCUSSION TITLE
extending spatial structures in MoBi

### DISCUSSION THREAD

**Discussion by csungeun** [Open-Systems-Pharmacology/Forum#881] [Questions & Problems] [answered]
I want to expand the lung model.
I referenced the paper(Applied Concepts in PBPK Modeling: How to Extend an Open Systems Pharmacology Model to the Special Population of Pregnant Women(2018)) but failed.

help me......
[lung model.zip](https://github.com/Open-Systems-Pharmacology/Forum/files/6219866/lung.model.zip)

**Comment by AndreDlm** [comment, marked as answer]
Dear @csungeun,

the errors tell you what went wrong - you can display them by displaying the `Notification` panel:
![image](https://user-images.githubusercontent.com/26541603/112858238-f93bf780-90b1-11eb-8ab2-b4881a95a4a5.png)

The first error tells you that one input variable needed for the the reaction via the unknown liver enzyme is missing (namely the `Partition coefficient (water/container)` in the compartment `ELF` in the lung). 
In the current implementation, this reaction occurs in all compartments where the unknown liver enzyme is present. This is defined in the MSV building block:
![image](https://user-images.githubusercontent.com/26541603/112858728-6e0f3180-90b2-11eb-9084-158305350807.png)

When setting up a new simulation, this enzyme is per default also present in the new lung compartment. So to solve this you would either need to add the parameter `Partition coefficient (water/container)` in all sub-compartments of the new the lung or you define in the MSV building blocks that the unknown liver enzyme is **not** present in these compartments. To do so, you need to switch to the MSV building block, click on `Extend` in the ribbon bar menu and deselect the unknown liver enzyme in all compartments of `nLung`:

![image](https://user-images.githubusercontent.com/26541603/112859493-2fc64200-90b3-11eb-8883-3d25d7da1e0f.png)

![image](https://user-images.githubusercontent.com/26541603/112859301-fd1c4980-90b2-11eb-9c30-09e55ab796fc.png)

The second error in the model is that the transport equation PassiveDiffusionInt2Cell refers to a parameter that cannot be found, namely the `Partition coefficient (interstitial/water)` under the path `Neighborhoods|nLung_int_nLung_cell|Digoxin|Partition coefficient (interstitial/water)`. To solve this, you just have to navigating there and correct the typos, then it should work: 
![image](https://user-images.githubusercontent.com/26541603/112859937-a2cfb880-90b3-11eb-92e5-8b4c7517eb17.png)

I hope this helps

**Comment by csungeun** [comment]
나는 당신의 대답에 감동했습니다. 대단히 감사합니다.
